src/main/java/com/google/api/codegen/discovery/transformer/java/JavaSampleTypeNameConverter.java

```diff
@@ -108,12 +97,25 @@ public TypeName getTypeNameForElementType(TypeInfo typeInfo) {
           keyTypeName,
           valueTypeName);
     }
-    return getTypeNameForElementType(typeInfo, true);
+    if (typeInfo.isArray()) {
+      TypeName listTypeName = typeNameConverter.getTypeName("java.util.List");
+      TypeName elementTypeName = getTypeNameForElementType(typeInfo, true);
+      return new TypeName(
+          listTypeName.getFullName(), listTypeName.getNickname(), "%s<%i>", elementTypeName);
+    }
+    return getTypeNameForElementType(typeInfo, false);
   }
 
-  private TypeName getNonMessageTypeName(TypeInfo typeInfo) {
+  @Override
+  public TypeName getTypeNameForElementType(TypeInfo typeInfo) {
```

Comment:
I think this name is still consistent (it was less appropriate before...) in that for any type you pass it in it returns the TypeName of its element.

A map type -> a map entry
An array type - > an array entry
A message -> a message
A primitive -> a boxed primitive